radio/src/hal/adc_driver.cpp

```diff
@@ -57,6 +60,12 @@ static bool adcSingleRead()
   if (_hal_adc_driver->wait_completion)
     _hal_adc_driver->wait_completion();
 
+  // Need to put all in a group to ensure DMA transfer will not affect ADC sampling
+#if defined(FLYSKY_GIMBAL) && !defined(SIMU)
+  flysky_gimbal_start_read();
+  flysky_gimbal_wait_completion();
+#endif  
+  
```

Comment:
> Need to put all in a group to ensure DMA transfer will not affect ADC sampling

What does this mean exactly?

Besides, I'd really prefer find a way to somehow register multiple such drivers, and execute them here like:
```c
for (unsigned i; i < n_drivers; i++) {
  drivers[i]->start_read();
}

for (unsigned i; i < n_drivers; i++) {
  drivers[i]->wait_completion();
}
```

Reply:
The best for timing is to fo flysky gimbal start read -> adc start read -> wait for both adc and flysky gimbal finished, however, not sure why when I put the call like this, the ADC output will jump, and can be observed in the channel monitor even you do nothing to the hardware.

Reply:
Didn't X12S uses PWM gimbals with SPI ADC?

Reply:
> Didn't X12S uses PWM gimbals with SPI ADC?

Not that I recall, it uses analog output HAL gimbal to and external ADC converter, not PWM

Reply:
> The best for timing is to fo flysky gimbal start read -> adc start read -> wait for both adc and flysky gimbal finished, however, not sure why when I put the call like this, the ADC output will jump, and can be observed in the channel monitor even you do nothing to the hardware. BTW, not sure if this is due to memory corruption similar to SPI's problem, the observation just not making any sense unless there are something nasty.

Interesting! This smells like we don't have the full picture. Something fishy must be going on... Dig deeper!

Reply:
Seems it is not easy to eliminate the loop waiting ways for synchronization without restructuring and split the mixer task into 2 different tasks: one obtain the data, and the other consume the data , do mixing and output.  Well, maybe delay to future refactoring.  The current implementation is just work at this moment.

And this is a precedence of the upcoming PA01, so I think it is better to merge it now, if this did not have any other side effect.